Description: Makefile: build before install
 Makefile adjusted to build in 'make' before 'make install'.
 .
 bonescript (0.5.0-1~bpo80+20160629+1) jessie; urgency=low
 .
   * Initial release.
Author: Jason Kridner <jdk@ti.com>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- bonescript-0.5.0.orig/Makefile
+++ bonescript-0.5.0/Makefile
@@ -1,5 +1,5 @@
 PREFIX:=/usr/local
-npm_options:=--unsafe-perm=true --progress=false --loglevel=error --prefix $(DESTDIR)$(PREFIX)
+npm_options:=--unsafe-perm=true --progress=false --loglevel=error
 NODE_PATH:=$(DESTDIR)$(PREFIX)/lib/node_modules
 NODEJS:=nodejs
 NPM:=npm
@@ -7,37 +7,34 @@ NPM:=npm
 ver=$(shell nodejs -pe "require('./package.json').version")
 npm_version=$(shell $(NPM) --version)
 node_version=$(shell $(NODEJS) --version)
+bs_path=./build/lib/node_modules/bonescript
 
 all:
-	@echo "ver=$(ver)"
-	@echo "npm_version=$(npm_version)"
-	@echo "node_version=$(node_version)"
+	mkdir -p $(bs_path)
+	echo $(ver) > $(bs_path)/bonescript.version
+	echo $(node_version) > $(bs_path)/bonescript.node_version
+	echo $(npm_version) > $(bs_path)/bonescript.npm_version
+	tar xf debian/bonescript-0.5.0-v0.12.15.tar.xz -C $(bs_path)/
+	cp -dr --preserve=mode,timestamp etc/* ./build/etc/
 
 clean:
-
+	rm -rf build
 
 test:
 
 
 install:
-	NODE_PATH=$(NODE_PATH) TERM=dumb $(NPM) install -g $(npm_options)
-	echo $(ver) > $(NODE_PATH)/bonescript/bonescript.version
-	echo $(node_version) > $(NODE_PATH)/bonescript/bonescript.node_version
+	install -m 0755 -d $(DESTDIR)$(PREFIX)
+	cp -dr --preserve=mode,timestamp ./build/* $(DESTDIR)$(PREFIX)/
 	systemctl stop bonescript-autorun.service || true
 	systemctl stop bonescript.service || true
 	systemctl stop bonescript.socket || true
-	install -m 0755 -d $(DESTDIR)/etc/default
-	install -m 0644 etc/default/node $(DESTDIR)/etc/default/
-	install -m 0755 -d $(DESTDIR)/etc/profile.d
-	install -m 0755 etc/profile.d/node.sh $(DESTDIR)/etc/profile.d/
 	install -m 0755 -d $(DESTDIR)/lib/systemd/system
 	install -m 0644 systemd/bonescript.socket $(DESTDIR)/lib/systemd/system/
 	install -m 0644 systemd/bonescript.service $(DESTDIR)/lib/systemd/system/
 	install -m 0644 systemd/bonescript-autorun.service $(DESTDIR)/lib/systemd/system/
 	systemctl enable bonescript.socket || true
 	systemctl enable bonescript.service || true
 	systemctl enable bonescript-autorun.service || true
-	install -m 0755 -d $(DESTDIR)/etc/avahi/services
-	install -m 0644 etc/avahi/services/bone101.service $(DESTDIR)/etc/avahi/services/
 
 .PHONY: clean test install
